modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/TableCell.tsx

```diff
@@ -12,17 +12,60 @@
  * details.
  */
 
-import React, {useContext, useEffect, useState} from 'react';
+import {FDSCellRenderer, FDSCellRendererArgs} from '@liferay/js-api/data-set';
+import React, {RefObject, useContext, useEffect, useState} from 'react';
+
+// @ts-ignore
 
 import FrontendDataSetContext from '../../FrontendDataSetContext';
+
+// @ts-ignore
+
 import DefaultRenderer from '../../data_renderers/DefaultRenderer';
 import {
+	AnyDataRenderer,
 	getDataRendererById,
 	getDataRendererByURL,
 	getInputRendererById,
 } from '../../utils/dataRenderers';
+
+// @ts-ignore
+
 import DndTableCell from './dnd_table/Cell';
 
+interface ClientExtensionRendererComponentProps {
+	args: FDSCellRendererArgs;
+	renderer: FDSCellRenderer;
+}
+
+class ClientExtensionRendererComponent extends React.Component<
+	ClientExtensionRendererComponentProps
+> {
+	private readonly ref: RefObject<HTMLDivElement>;
+
+	constructor(props: ClientExtensionRendererComponentProps) {
+		super(props);
+
+		this.ref = React.createRef();
+	}
+
+	render() {
+
+		// By the time render is invoked for the first time the ref is not yet
+		// populated, so we need to defer the rendering
+
+		setTimeout(() => {
+			if (this.ref.current) {
+				this.ref.current.appendChild(
+					this.props.renderer(this.props.args)
+				);
+			}
+		}, 0);
```

Comment:
Most probably :+1:  What happens here is:

1. I'm not an expert in React
2. I had to use setTimeout(0) to make it schedule for the next tick. If I used setTimeout(1) it lead to weird issues due to FDS rendering synchronization. However, I've changed the code several times so it would be good to check again, as you say.

Reply:
I've pushed a change using `componentDidMount` :+1: